zapisy/apps/enrollment/courses/views.py

```diff
@@ -129,23 +130,51 @@ def group_view(request, group_id):
     except Group.DoesNotExist:
         raise Http404
 
-    records = Record.objects.filter(
-        group_id=group_id).exclude(status=RecordStatus.REMOVED).select_related(
+    order = Func(
+        'student__user__last_name',
+        function='pl_PL',
+        template='(%(expressions)s) COLLATE "%(function)s"')
+
+    """
+        ORDER BY will sort records depending on the database locale (collation).
+        We can either make sure that database uses the locale we need or
+        ask for proper collation in sql queries.
+
+        In this case, we simply ask our database (through Django) to run a query:
+        SELECT ... FROM ...
+        .
+        .
+        .
+        ORDER BY student__user__last_name COLLATE pl_PL
+
+        It will work in any database supporting COLLATE (both PostgreSQL and MySQL do).
+    """
+
+    records_group = Record.objects.filter(
+        group_id=group_id, status=RecordStatus.ENROLLED).select_related(
+            'student', 'student__user', 'student__program',
+            'student__consent').prefetch_related('student__user__groups').order_by(order)
+
+    records_queue = Record.objects.filter(
+        group_id=group_id, status=RecordStatus.QUEUED).select_related(
             'student', 'student__user', 'student__program',
             'student__consent').prefetch_related('student__user__groups').order_by('created')
 
     guaranteed_spots_rules = GuaranteedSpots.objects.filter(group=group)
 
     students_in_group = []
     students_in_queue = []
-    record: Record
-    for record in records:
-        record.student.guaranteed = set(rule.role.name for rule in guaranteed_spots_rules) & set(
-            role.name for role in record.student.user.groups.all())
-        if record.status == RecordStatus.ENROLLED:
-            students_in_group.append(record.student)
-        elif record.status == RecordStatus.QUEUED:
-            students_in_queue.append(record.student)
+
+    def handle_records(records, student_list):
+        record: Record
+        for record in records:
+            record.student.guaranteed = set(rule.role.name for rule in guaranteed_spots_rules) & set(
+                role.name for role in record.student.user.groups.all())
+            student_list.append(record.student)
+
+    handle_records(records_group, students_in_group)
```

Comment:
To wygląda bardzo w stylu C.
W Pythonie można zwracać listy.

```suggestion
    students_in_group = collect_students(records_group)
```